covid_moonshot_ml/plotting/plotting.py

```diff
@@ -0,0 +1,348 @@
+import numpy as np
+import math
+import pandas as pd
+import plotly.express as ex
+
+
+class Rock:
+    """
+    Class for getting ROC-AUC results for different scoring functions for a dataset.
+    """
+
+    def __init__(
+        self,
+        df,
+        score_name,
+        rmsd_name,
+        n_samples,
+    ):
+        self.df = df
+        self.score_name = score_name
+        self.rmsd_name = rmsd_name
+        self.n_samples = n_samples
+        self.get_score_range()
+
+        (
+            self.total_poses,
+            self.total_good_poses,
+            self.total_bad_poses,
+            self.total_cmpds,
+            self.total_good_cmpds,
+            self.total_bad_cmpds,
+        ) = self.calc_data(self.df)
+
+        self.auc = str
+        self.auc_list = []
+
+    def calc_data(self, df):
+        """
+        Given a dataset, return the six things we want to know
+        Parameters
+        ----------
+        df
+
+        Returns
+        -------
+
+        """
+        n_poses = len(df)
+        n_good_poses = sum(df[self.rmsd_name] <= 2)
+        n_bad_poses = n_poses - n_good_poses
+
+        n_cmpds = len(set(df.Compound_ID))
+        set_of_good_cmpds = set(df[df[self.rmsd_name] <= 2].Compound_ID)
+        n_good_cmpds = len(set_of_good_cmpds)
+        n_bad_cmpds = n_cmpds - n_good_cmpds
+
+        return (
+            n_poses,
+            n_good_poses,
+            n_bad_poses,
+            n_cmpds,
+            n_good_cmpds,
+            n_bad_cmpds,
+        )
+
+    def calc_auc(self, false_positive_rates, true_positive_rates):
+        """
+        Calculates the area under the curve using the numpy function!
+        Parameters
+        ----------
+        false_positive_rates
+        true_positive_rates
+
+        Returns
+        -------
+
+        """
+        return np.trapz(x=false_positive_rates, y=true_positive_rates)
+
+    def get_score_range(self):
+        """
+        For some score in the data with arbitrary minima and maxima, return n evenly spaced points along the score.
+        Returns
+        -------
+
+        """
+        self.score_range = np.linspace(
+            self.df[self.score_name].min() - 1,
+            self.df[self.score_name].max(),
+            self.n_samples,
+            endpoint=True,
+        )
+
+    def weird_division(self, n, d):
+        return n / d if d else 0
+
+    def get_auc_from_df(self, df=None, bootstrap=False):
+        if df is None:
+            ## In the case that we're using the data saved to this class, we can pull out the original totals
+            df = self.df
+            total_poses = self.total_poses
+            total_good_poses = self.total_good_poses
+            total_bad_poses = self.total_bad_poses
+            total_good_cmpds = self.total_good_cmpds
+            total_bad_cmpds = self.total_bad_cmpds
+
+        else:
```

Comment:
consider test for if `df` is indeed a dataframe.

Reply:
added here: https://github.com/choderalab/covid-moonshot-ml/pull/39/commits/53dcf89adb8e0619c4494aa3dc2775aeac34acaa